Minor tweaks to readme to bring it more inline with actuality and re… (#8)
opencredo/kafka-connect-venafi-tpp: README.md

kafka-connect-venafi-tpp is a [Kafka connector](http://kafka.apache.org/documentation.html#connect) for Venafi Trust Protection Platform security events.

Description
---

> This connector connects via HTTP to your instance of the Venafi Trust Protection Platform ( which shall be refered to TPP from here on) and pulls your Log events into kafka allowing you to do any filtering/transforming/processing you'd like to do within a comfortable kafka environment.  

Reviewer: connects to Foo via Bar

Author: I think both are valid.

> This connector connects via HTTP to your instance of the Venafi Trust Protection Platform ( which shall be refered to TPP from here on) and pulls your Log events into kafka allowing you to do any filtering/transforming/processing you'd like to do within a comfortable kafka environment.  

Reviewer: ```suggestion
This connector connects via HTTP to your instance of the Venafi Trust Protection Platform ( which shall be referred to as TPP from here on) and pulls your Log events into Kafka, allowing you to do any filtering/transforming/processing you'd like to do within a comfortable Kafka environment.  
```

Reviewer: Is "comfortable" the right word?

>  **N.B.** Currently the connector starts from the of the beginning of time, a future release will allow the option of starting from now.

Reviewer: ```suggestion
 **N.B.** Currently the connector starts from the beginning of time (i.e. processes all past events first), a future release will allow the option of starting from now (i.e. skipping all previous events).
```

> EventLog object produced by this source **Connector** as defined by the Venafi WebSDK

Reviewer: ```suggestion
**EventLog**: object produced by this source Connector as defined by the Venafi WebSDK.
```

Author: The line is meant to be more of a header.

> * **ClientTimestamp**: The time that the client generated the event.

Reviewer: ```suggestion
* **ClientTimestamp**: The time that the client generated the event at.
```
Check with another native speaker. Cf. this article (esp. page 2) https://www.quickanddirtytips.com/education/grammar/ending-a-sentence-with-a-preposition?page=1. Part of me wants a preposition in that sentence, part of me thinks it's redundant, as in "The time the client called me." - maybe `The time at which the client generated the event.`?

Author: This is taken pretty much verbatim from the documentation, so...

> * **Component**: A string that identifies a component DN. For events that occur in a subsystem other than Config, such as a Secret Store, the Distinguished Name (DN) component may be blank.

Reviewer: ```suggestion
* **Component**: A string that identifies a component Distinguished Name (DN). For events that occur in a subsystem other than Config, such as a Secret Store, the Distinguished Name component may be blank.
```

* **ComponentId**: The component ID that originated the event.
* **ComponentSubsystem**: The component subsytem that originated the event.

> * **Grouping**: An integer for the component ID or Group Id that correlates to a set of events.

Reviewer: ```suggestion
* **Grouping**: An integer for the component ID or Group ID that correlates to a set of events.
```

* **Id**: A hexadecimal value that corresponds to an Event ID.
* **Name**: The event name.
* **ServerTimestamp**: The time the Trust Protection Platform server received the event.
* **Severity**: A string value for one of these event severity: Emergency, Alert, Critical, Error, Warning, Notice, Info, or Debug
* **SourceIP**: The IP address of the host that originated the event
* **Text1**: A string variable, that contains all or part of an event message.

> * **Text2**: An additional string if present, that appears as part of an event message.

Reviewer: ```suggestion
* **Text2**: An additional string, that, if present, appears as part of an event message.
```
NB: I'm not a native speaker, so chances are my gut feeling here is 100% wrong... ;-)

connector.class=com.opencredo.connect.venafi.tpp.log.TppLogSourceConnector
tasks.max=1
type.name=kafka-connect
venafi.base.url=<your_base_url>
venafi.username=<your_api_username>
venafi.password=<your_api_password>
```
This is filled with the minimum values required, any default values are provided by the [config definition class](./src/main/java/com/opencredo/connect/venafi/tpp/log/TppLogSourceConfig.java). 

> This can also be looked at for more information on configuration or look at the [wiki on the config definitions.](https://github.com/opencredo/kafka-connect-venafi-tpp/wiki/Config-Definitions-explained.)

Reviewer: ```suggestion
This can also be looked at for more information on configuration, or look at the [wiki on the config definitions.](https://github.com/opencredo/kafka-connect-venafi-tpp/wiki/Config-Definitions-explained.)
```
I think? (Enumeration?)

> Make sure to replace the items with the <> brackets with what's needed to connect to your Venafi TPP/VEDSDK instance.   

Reviewer: Check the pointy braces actually render in markdown

> 4. Create a directory to place the JAR file in and place it there e.g. `<path-to-confluent>/share/java/kafka-connect-venafi`.  

Reviewer: ```suggestion
4. Create a directory and place the JAR file in it, e.g. `<path-to-confluent>/share/java/kafka-connect-venafi`.  
```

5. Then call: `<path-to-confluent>/bin/confluent load venafi -d venafi.properties`  

For more information please look at the [Confluent instructions on manually installing connectors](https://docs.confluent.io/current/connect/managing/install.html#connect-install-connectors).

---

> If you need to unload/reload it use: `<path-to-confluent>/bin/confluent unload venafi`  

Reviewer: ```suggestion
If you need to unload/reload the connector use: `<path-to-confluent>/bin/confluent unload venafi`  
```
comma before "use"? I think I'm currently mixing German and English rules, and it's not helping... ;-)

Author: I mean I think you're right.

sudo bin/kafka-avro-console-consumer --bootstrap-server localhost:9092 --topic TPP-LOGS
sudo bin/confluent log connect
``` 
---

# Config Definitions explained.

``venafi.base.url``

>   Url to TPP api with /VEDSDK

Reviewer: ```suggestion
  URL to TPP API, usually ends in `/VEDSDK`
```

Author: Done

  * Type: string
  * Valid Values: non-empty string and no ISO control characters
  * Importance: high

``venafi.password``
  The password to use with the /VEDSDK api.

  * Type: string
  * Importance: high

``venafi.username``
  The username to use with the /VEDSDK api.

  * Type: string
  * Importance: high

``venafi.topic``

>   TOPIC to publish TPP log data to.

Reviewer: ```suggestion
  Topic to publish TPP log data to.
```
Use bold to emphasise, not caps?

Author: Done

  * Type: string
  * Default: TPP-LOGS
  * Valid Values: non-empty string and no ISO control characters
  * Importance: high

``venafi.batch.size``

>   Window of data to pull from log api.

Reviewer: ```suggestion
  Window of data to pull from log API.
```
